uniffi_bindgen/src/interface/types/codetype.rs

```diff
@@ -0,0 +1,210 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
```

Comment:
How would you feel about moving this into the `::bindings::backend` module (or submodules thereof) rather than having it live here under `::interface`? It's code for the "output" phase rather than the "input" phase of the tool, and I feel like folks are more likely to look for such code over in `::bindings` somewhere.

Organizing the code so that all the generic helpers for writing a backend are under `::bindings::backend` also gives us a single obvious place to write high-level docs about how to build a backend - they can be in the module-level doc comments for that generic helper code.

OTOH, maybe that would make it a bit weird for the scaffolding code to import and use it. A shared `::codegen` module maybe? Somehow, it feels important enough that it shouldn't be hidden away inside `::interface` with a bunch of internal implementation details of the parsing code.